### ISSUE TITLE
Cannot authenticate to Bitbucket Server using personal access token

### ISSUE THREAD

**Issue by doxxx** [sourcegraph/sourcegraph#221]
- **Sourcegraph version:** 2.11.2
- **Platform information:** 
  - Ubuntu 16.04
  - Docker 18.06.1-ce
  - Bitbucket Server 5.13.1

#### Steps to reproduce:

1. Start Sourcegraph Docker container as per Quickstart instructions.
2. Add Bitbucket Server repo to Sourcegraph config with https URL and personal access token generated for my Bitbucket user (with admin rights, can see all repos).
3. Enable a repository.

#### Expected behavior:

Repository can be mirrored and indexed.

#### Actual behavior:

The Repository Mirror settings page says:

```
The remote repository is unreachable. Logs follow.

repo not found (name=git.synaptivemedical.com/GLIA/glia url=https://git.synaptivemedical.com/scm/glia/glia.git notfound=false) because exit status 128 (output follows)

fatal: Authentication failed for 'https://git.synaptivemedical.com/scm/glia/glia.git/'
```

On the Docker host and inside the Sourcegraph container, I can `git clone` the repo when I add my username and personal access token to the repo URL. When I checked the Bitbucket Server REST API that lists repos, none of the http clone URLs include the username, but the [Sourcegraph code appears to be assuming that it does](https://github.com/sourcegraph/sourcegraph/blob/16d03a7c5641d41b587af4db04ca33905bc2455d/cmd/repo-updater/repos/bitbucketserver.go#L103).

**Comment by keegancsmith** [comment]
It sounds like we need to add a username field to the bitbucket config. Then when creating the SSH or HTTP(s) URL we include the username if it is set. Does that sound correct to you @doxxx?

**Comment by doxxx** [comment]
The Bitbucket config already has a username field but there is validation that fails when both a username and token field are provided. I think if that validation was relaxed and the username used if present, it should fix the problem.

I haven't tried using a personal access token with SSH in Bitbucket, so I can't say whether the username is necessary.

**Comment by joaoqalves** [comment]
I ran into the same situation. For the time being, I'm just using SSH

**Comment by keegancsmith** [comment]
Great I'll look into a PR for this tomorrow. Thanks for the information.

**Comment by keegancsmith** [comment]
Sorry I didn't get time to look into this today. Monday expect a PR :)

**Comment by doxxx** [comment]
Thanks for the fix! How can I determine which version (Docker image) will include the fix?

**Comment by keegancsmith** [comment]
Sorry didn't mean to auto-close. The `insiders` tag for the docker image will contain the fix, but that follows master so is not a stable tag to rely on. We will be doing a release soon which should contain the fix, and that would be a better tag to pin on.

**Comment by doxxx** [comment]
I just tried the latest insiders tag (sha256:2bf1d3bf3335ce1d3d11e705cfd641760ff75cc00bc34887a6f791e21e215bc8) but I'm still getting an error, when configuring the Bitbucket connection, which says that I cannot specify a username as well as a personal token.

**Comment by keegancsmith** [comment]
argh so sorry, will fix that right now (oversight on my part)

**Comment by keegancsmith** [comment]
Fixed by https://github.com/sourcegraph/sourcegraph/commit/70d4471b80e791e0b2eaeb39d7c1045ddffc37a5

There is a new image. Image digest: `sha256:d407bb6ea6fc355a8059e930335a61c389e4b7cde39df0661d2c62d7a550e3b8`. Tag `sourcegraph/server:01195_2018-10-14_23150ce`. Or just use `:insiders` as mentioned above.

**Comment by doxxx** [comment]
I can confirm that this works with the latest insiders tag.

**Comment by nicksnyder** [comment]
This is fixed in 2.12.0
